site-packages/integralstor/rsync.py

```diff
@@ -0,0 +1,340 @@
+                that are rsync command ready
+    """
+
+    switches_formed = {}
+    switches_formed['short'] = None
+    switches_formed['long'] = None
+
+    try:
+        sw_long = ''
+        sw_short = ''
+        if not switches:
+            raise Exception('Malformed switches')
+
+        for k, v in switches.items():
+            if v['is_long']:
+                if v['is_arg'] and v['arg_value']:
+                    val = '%s="%s"' % (str(v['key']), str(v['arg_value']))
+                else:
+                    val = '%s' % str(v['key'])
+                sw_long = '%s %s' % (sw_long, val)
+            else:
+                if v['is_arg'] and v['arg_value']:
+                    val = '%s="%s"' % (str(v['key']), str(v['arg_value']))
+                else:
+                    val = '%s' % str(v['key'])
+                sw_short = '%s %s' % (sw_short, val)
+
+        switches_formed['short'] = sw_short.strip()
+        switches_formed['long'] = sw_long.strip()
+
+    except Exception, e:
+        return None, str(e)
+    else:
+        return switches_formed, None
+
+
+def _generate_rsync_config():
+    """
+    This generates the rsync config.
+    Assumed is that, xinetd is already installed and rsync deamon is enabled in xinetd
+
+    Path of config : /etc/rsyncd.conf
+
+    Default Config :
+    [rsync_share_name]
+
+    path=/pool/dataset/dir
+    comment="Description of this share"
+    list=true
+    readonly = False, Selectable
+    uid = integralstor (Selectable)
+    gid = integralsto (Selectable)r
+
+    """
+
+    try:
+        shares, err = load_shares_list()
+        if err:
+            raise Exception(err)
+        with open("/etc/rsyncd.conf", "w") as f:
+            f.write("uid = integralstor\n")
+            f.write("gid = integralstor\n")
+            f.write("\n")
+            for share in shares:
+                f.write("[%s]\n" % share["name"])
+                f.write("\n")
+                f.write("path = %s\n" % share["path"])
+                f.write("comment = %s\n" % share["comment"])
+                list = 'true' if share["list"] else 'false'
+                readonly = 'true' if share["readonly"] else 'false'
+                f.write("list = %s\n" % list)
+                f.write("readonly = %s\n" % readonly)
+                f.write("\n")
+    except Exception, e:
+        return False, 'Error generating rsync config : %s' % str(e)
+    else:
+        return True, None
+
+
+def create_rsync_share(name, path, comment, list, readonly, uid, gid):
+    try:
+        db_path, err = config.get_db_path()
+        if err:
+            raise Exception(err)
+
+        check, err = db.get_single_row(
+            db_path, "select * from rsync_shares where name='%s'" % name)
+        if check:
+            raise Exception("Share already exists.Use a different share name")
+
+        cmd_list = []
+        cmd = ["insert into rsync_shares (name,path,comment,list,readonly,uid,gid) values(?,?,?,?,?,?,?)", (
```

Comment:
I had noticed that the rsunc_shares table does not contain a unique autoincrement ID field.. Shd we put that in now?